modules/java/generator/src/cpp/listconverters.cpp

```diff
@@ -0,0 +1,55 @@
+#define LOG_TAG "org.opencv.utils.Converters"
+#include "common.h"
+
+#define JAVA_UTIL_ARRAYLIST "java/util/ArrayList"
+
+
+jobject vector_String_to_List(JNIEnv* env, std::vector<cv::String>& vs) {
+
+    static jclass juArrayList       = static_cast<jclass>(env->NewGlobalRef(env->FindClass(JAVA_UTIL_ARRAYLIST)));
+    static jmethodID m_create   = env->GetMethodID(juArrayList, "<init>", "(I)V");
+    jmethodID m_add       = env->GetMethodID(juArrayList, "add", "(Ljava/lang/Object;)Z");
+
+    jobject result = env->NewObject(juArrayList, m_create, vs.size());
+    for (std::vector<cv::String>::iterator it = vs.begin(); it != vs.end(); ++it) {
+        jstring element = env->NewStringUTF((*it).c_str());
+        env->CallBooleanMethod(result, m_add, element);
+        env->DeleteLocalRef(element);
+    }
```

Comment:
It can be but the first  vector_String_to_List is creating the list while Copy_vector_String_to_List is for updating the list in case of CV_OUT parameter. IMHO minor issue, we can refactor it later if needed.